common/alerts.js

```diff
@@ -52,5 +52,21 @@
                 };
             }
         };
-    }]);
+    }])
+    .directive('compile', ['$compile', function ($compile) {
+      return function(scope, element, attrs) {
+        scope.$watch(
+          function(scope) {
+            return scope.$eval(attrs.compile);
+          },
+          function(value) {
+            element.html(value);
```

Comment:
What is the evaluated value here? Is it the HTML string we passed to the attribute? If that's true, is this replacing the HTML for the element (which is fine in this case because the span tag is meant to only contain the content we are trying to compile here)?